go/libraries/doltcore/dbfactory/oss.go

```diff
@@ -0,0 +1,74 @@
+package dbfactory
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/aliyun/aliyun-oss-go-sdk/oss"
+	"github.com/dolthub/dolt/go/store/blobstore"
+	"github.com/dolthub/dolt/go/store/chunks"
+	"github.com/dolthub/dolt/go/store/datas"
+	"github.com/dolthub/dolt/go/store/nbs"
+	"github.com/dolthub/dolt/go/store/prolly/tree"
+	"github.com/dolthub/dolt/go/store/types"
+	"net/url"
+	"os"
+)
+
+const (
+	ossEndpointEnvKey        = "OSS_ENDPOINT"
+	ossAccessKeyIDEnvKey     = "OSS_ACCESS_KEY_ID"
+	ossAccessKeySecretEnvKey = "OSS_ACCESS_KEY_SECRET"
```

Comment:
If you use environment variables it means that dolt can only have a single oss remote.  It would be better if the user was able to setup multiple oss remotes.